src/main/java/br/com/six2six/fixturefactory/transformer/DateTimeTransformer.java

```diff
@@ -1,51 +1,74 @@
-			
-		} else if (ClassUtils.isAssignable(type, java.time.LocalTime.class)) {
-			returnValue = toZonedDateTime((Calendar) value).toLocalTime();
-			
-		} else if (ClassUtils.isAssignable(type, java.time.OffsetDateTime.class)) { 
-			returnValue = toZonedDateTime((Calendar) value).toOffsetDateTime();
-			
-		} else if (ClassUtils.isAssignable(type, java.time.ZonedDateTime.class)) { 
-			returnValue = toZonedDateTime((Calendar) value);
+		try {
+			Object zonedDateTime = DateTimeUtils.toZonedDateTime((Calendar) value);
+			if (ClassUtils.isAssignable(type, ClassUtils.getClass("java.time.LocalDateTime"))) {
+				returnValue = invokeMethod(zonedDateTime, "toLocalDateTime", EMPTY_OBJECT_ARRAY);
+				
+			} else if (ClassUtils.isAssignable(type, ClassUtils.getClass("java.time.LocalDate"))) {
+				returnValue = invokeMethod(zonedDateTime, "toLocalDate", EMPTY_OBJECT_ARRAY);
+				
+			} else if (ClassUtils.isAssignable(type, ClassUtils.getClass("java.time.LocalTime"))) {
+				returnValue = invokeMethod(zonedDateTime, "toLocalTime", EMPTY_OBJECT_ARRAY);
+				
+			} else if (ClassUtils.isAssignable(type, ClassUtils.getClass("java.time.OffsetDateTime"))) {
+				returnValue = invokeMethod(zonedDateTime, "toOffsetDateTime", EMPTY_OBJECT_ARRAY);
+				
+			} else if (ClassUtils.isAssignable(type, ClassUtils.getClass("java.time.ZonedDateTime"))) { 
+				returnValue = zonedDateTime;
+				
+			} else if (ClassUtils.isAssignable(type, ClassUtils.getClass("java.time.OffsetTime"))) { 
+				returnValue = invokeMethod(zonedDateTime, "toOffsetDateTime", EMPTY_OBJECT_ARRAY);
+				returnValue = invokeMethod(returnValue, "toOffsetTime", EMPTY_OBJECT_ARRAY);
+				
+			} else if (ClassUtils.isAssignable(type, ClassUtils.getClass("java.time.Instant"))) { 
+				returnValue = invokeMethod(zonedDateTime, "toInstant", EMPTY_OBJECT_ARRAY);
+				
+			} else {
+				throw new IllegalArgumentException("Incorrect type for transformer: " + type.getCanonicalName());
+			}
 			
-		} else if (ClassUtils.isAssignable(type, java.time.OffsetTime.class)) { 
-			returnValue = toZonedDateTime((Calendar) value).toOffsetDateTime().toOffsetTime();
-			
-		} else if (ClassUtils.isAssignable(type, java.time.Instant.class)) { 
-			returnValue = ((Calendar) value).toInstant();
-			
-		} else {
-			throw new IllegalArgumentException("Incorrect type for transformer: " + type.getCanonicalName());
+		} catch (NoSuchMethodException e) {
+			throw new IllegalArgumentException("Problem to convert to type: " + type.getCanonicalName(), e);
+		} catch (IllegalAccessException e) {
+			throw new IllegalArgumentException("Problem to convert to type: " + type.getCanonicalName(), e);
+		} catch (InvocationTargetException e) {
+			throw new IllegalArgumentException("Problem to convert to type: " + type.getCanonicalName(), e);
+		} catch (ClassNotFoundException e) {
+			throw new IllegalArgumentException("Problem to convert to type: " + type.getCanonicalName(), e);
```

Comment:
This will work with the java compiler 6?
Because I knew how to do but did not because of it.